### DISCUSSION TITLE
Realtime Unhandled Exception in flutter

### DISCUSSION THREAD

**Discussion by AkmenZ** [supabase/supabase#18034] [Questions]
Hello. 
I'm building a flutter app that has chat function as well, and i notice this error in debug console:

[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception:
#0      SupabaseStreamBuilder._addException
supabase_stream_builder.dart:442
#1      SupabaseStreamBuilder._getStreamData.<anonymous closure>
supabase_stream_builder.dart:348
#2      RealtimeChannel.subscribe.<anonymous closure>
realtime_channel.dart:202
#3      RealtimeChannel.onError.<anonymous closure>
realtime_channel.dart:331
#4      RealtimeChannel.trigger
realtime_channel.dart:569
#5      RealtimeClient._triggerChanError
realtime_client.dart:415
#6      RealtimeClient._onConnClose
realtime_client.dart:396
#7      RealtimeClient.connect.<anonymous closure>
realtime_client.dart:154
#8      _RootZone.runGuarded (dart:async/zone.dart:1582:10)
#9      _BufferingStreamSubscription._sendDone.sendD<…>

This happens when laptop with simulator running is closed for about 5-10 minutes and the reopened. 
<img width="1084" alt="Screenshot 2023-10-09 at 12 54 13" src="https://github.com/supabase/supabase/assets/78148307/fb782ce3-cb68-4d3b-beab-ac24d470aae8">

In app nothing changes, the streams and messages work as they supposed to, just the console shows a number (like 7 or 8 of the same) of these errors.

Is this an issue that i need to address with some stream controller management, or is this just something Supabase handles internally and just shows that there was these errors?

**Comment by GaryAustin1** [comment]
I don't know about flutter, but realtime will error if it is in background tab or powered down after awhile as the clocks get slowed down and it misses heartbeats.  You have to restart realtime after any such event (I restart in js when the tab is visible again and there has been an error).  Restart includes reloading in potentially missed data.
